tests/test_jobs.py

```diff
@@ -0,0 +1,151 @@
+    assert len(jobs & {"status": "error"}) == 2
+    assert len(jobs & {"table_name": "a", "status": "error"}) == 1
+    jobs.delete()
+
+
+def test_sigint(schema_any):
+    try:
+        schema.SigIntTable().populate(reserve_jobs=True)
+    except KeyboardInterrupt:
+        pass
+
+    assert len(schema_any.jobs.fetch()), "SigInt jobs table is empty"
+    status, error_message = schema_any.jobs.fetch1("status", "error_message")
+    assert status == "error"
+    assert error_message == "KeyboardInterrupt"
+
+
+def test_sigterm(schema_any):
+    try:
+        schema.SigTermTable().populate(reserve_jobs=True)
+    except SystemExit:
+        pass
+
+    assert len(schema_any.jobs.fetch()), "SigTerm jobs table is empty"
+    status, error_message = schema_any.jobs.fetch1("status", "error_message")
+    assert status == "error"
+    assert error_message == "SystemExit: SIGTERM received"
+
+
+def test_suppress_dj_errors(schema_any):
+    """test_suppress_dj_errors: dj errors suppressible w/o native py blobs"""
+    with dj.config(enable_python_native_blobs=False):
+        schema.ErrorClass.populate(reserve_jobs=True, suppress_errors=True)
+    assert len(schema.DjExceptionName()) == len(schema_any.jobs) > 0
+
+
+def test_long_error_message(schema_any, subjects):
+    # create long error message
+    long_error_message = "".join(
+        random.choice(string.ascii_letters) for _ in range(ERROR_MESSAGE_LENGTH + 100)
+    )
+    short_error_message = "".join(
+        random.choice(string.ascii_letters) for _ in range(ERROR_MESSAGE_LENGTH // 2)
+    )
+    assert subjects
+    table_name = "fake_table"
+
+    key = subjects.fetch("KEY")[0]
+
+    # test long error message
+    schema_any.jobs.reserve(table_name, key)
+    schema_any.jobs.error(table_name, key, long_error_message)
+    error_message = schema_any.jobs.fetch1("error_message")
+    assert (
+        len(error_message) == ERROR_MESSAGE_LENGTH
+    ), "error message is longer than max allowed"
+    assert error_message.endswith(
+        TRUNCATION_APPENDIX
+    ), "appropriate ending missing for truncated error message"
+    schema_any.jobs.delete()
+
+    # test long error message
+    schema_any.jobs.reserve(table_name, key)
+    schema_any.jobs.error(table_name, key, short_error_message)
+    error_message = schema_any.jobs.fetch1("error_message")
+    assert error_message == short_error_message, "error messages do not agree"
+    assert not error_message.endswith(
+        TRUNCATION_APPENDIX
+    ), "error message should not be truncated"
+    schema_any.jobs.delete()
+
+
+def test_long_error_stack(schema_any, subjects):
+    # create long error stack
+    STACK_SIZE = (
+        89942  # Does not fit into small blob (should be 64k, but found to be higher)
+    )
+    long_error_stack = "".join(
+        random.choice(string.ascii_letters) for _ in range(STACK_SIZE)
+    )
+    assert subjects
+    table_name = "fake_table"
+
+    key = subjects.fetch("KEY")[0]
```

Comment:
you can limit in these cases for better performance: `subjects.fetch("KEY", limit=1)[0]`